### PR TITLE
cat-1 model for food item- initial changes

### PR DESCRIPTION
**Pull request by asifmons**

### CODE REVIEW

**Review by sijosaji — COMMENTED** [review]
Please find comments

File: caterer/src/main/java/com/vendor/caterer/interfaces/CatererMapper.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.vendor.caterer.dto.CatererCreateRequest;
+import com.vendor.caterer.dto.CatererUpdateRequest;
+import com.vendor.caterer.model.Caterer;
+import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.NullValuePropertyMappingStrategy;
+import org.mapstruct.factory.Mappers;
+
+@Mapper(componentModel = "spring", nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
```
Comment (sijosaji) [inline caterer/src/main/java/com/vendor/caterer/interfaces/CatererMapper.java:11]: Why you need component model?

File: caterer/src/main/java/com/vendor/caterer/model/FoodItem.java [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+@Document(indexName = "food-item")
+public class FoodItem {
+    private UUID id;
+    private String name;
+    private String description;
+    private BigDecimal price;
```
Comment (sijosaji) [inline caterer/src/main/java/com/vendor/caterer/model/FoodItem.java:15]: For case of drinks there might be different price for different volume so need handle that scenario for drinks

File: caterer/src/main/java/com/vendor/caterer/model/Menu.java
Comment (sijosaji) [inline caterer/src/main/java/com/vendor/caterer/model/Menu.java:1]: Didn't understood this model

**Review by sijosaji — COMMENTED** [review]
Pls see comments

File: caterer/src/main/java/com/vendor/caterer/interfaces/CatererMapper.java (hunk 1 above)
Comment (asifmons) [reply caterer/src/main/java/com/vendor/caterer/interfaces/CatererMapper.java:11]: @Mapper(componentModel = "spring") ensures MapStruct creates a Spring-managed bean for the mapper.

File: caterer/src/main/java/com/vendor/caterer/model/FoodItem.java (hunk 2 above)
Comment (asifmons) [reply caterer/src/main/java/com/vendor/caterer/model/FoodItem.java:15]: during the discussion, we have assumed that the price for every item (including drinks) will per person.

File: caterer/src/main/java/com/vendor/caterer/model/Menu.java
Comment (asifmons) [reply caterer/src/main/java/com/vendor/caterer/model/Menu.java:1]: need to discuss

**Review by sijosaji — APPROVED** [review]

**Pull request merged**